core/src/main/java/org/spongepowered/configurate/objectmapping/ObjectFieldDiscoverer.java

```diff
@@ -149,7 +150,7 @@ private void collectFields(final AnnotatedType clazz, final FieldCollector<Map<F
 
             field.setAccessible(true);
             final AnnotatedType fieldType = getFieldType(field, clazz);
-            fieldMaker.accept(field.getName(), fieldType, Types.combinedAnnotations(fieldType, field),
+            fieldMaker.accept(field.getName(), updateAnnotations(fieldType, field.getAnnotations()), Types.combinedAnnotations(fieldType, field),
```

Comment:
This merging step would have to be done on every FieldDiscoverer implementation individually -- maybe there's a more thorough way of implementing this, say by doing the annotation merge using the AnnotatedElement provided within the `fieldMaker` instead?

Otherwise this would require changes to both the kotlin data class discoverer, and the record discoverer

(in addition to the future constructor-based discoverer)

Reply:
Ok, I moved it to where the FieldData is created in the ObjectMapperFactoryImpl